tests/scripts/deploy-validate-vault.sh

```diff
@@ -124,7 +78,9 @@ EOF
 
 function deploy_vault {
   # TLS config
-  generate_vault_tls_config
+  scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
```suggestion
  scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
IMO you can directly call the file and pass the argument to keep it simple and this is what we have in other instances

Reply:
Updated like this.

Reply:
CI: is throwing error this way,
```
+ bash tests/scripts/generate-tls-config.sh /home/runner/work/rook/rook/tlscertdir932281739 rook-ceph-rgw-tlsteststore smoke-ns ''
bash: tests/scripts/generate-tls-config.sh: No such file or directory
    ceph_base_object_test.go:433: 
```

Reply:
take help from this https://github.com/rook/rook/blob/3d68f7e58a0f07a7af62e6265114df9a32cb169d/tests/scripts/github-action-helper.sh#L80

Reply:
this function is directly called from `canary-integration-test.yaml I see, 
in which they are at the directory location,
But for calling from the shell script we need a Parent Path for running, or I should have to give the whole path like this`parent path = /home/runner/work/rook/rook/tests`.

Going to the previous way I feel more convenient for now.